src/constructor/config_structure.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass, field
+from enum import Enum
+
+
+# Phase utils
+class Phase(Enum):
+    TRAIN = 'train'
+    VALID = 'valid'
+    TEST = 'test'
+    PREDICT = 'predict'
+
+phase_mapping = {
+    'train': Phase.TRAIN,
+    'valid': Phase.VALID,
+    'test': Phase.TEST,
+    'predict': Phase.PREDICT
+}
+
+
+# Optimization parameters
+@dataclass
+class OptmizerParams:
+    name: str
+    params: Dict = field(default_factory=dict)
+    
+@dataclass
+class SchedulerPLParams:
+    interval: Optional[str] = None
+    monitor: Optional[str] = None
+
+@dataclass
+class SchedulerParams:
+    name: str
+    pl_params: Optional[SchedulerPLParams] = None
+    params: Dict = field(default_factory=dict)
+
+@dataclass
+class OptimizationParams:
+    optimizer: OptmizerParams
+    scheduler: Optional[SchedulerParams]
+
+
+# Data parameters
+@dataclass
+class AugmentationParams:
+    name: str
+    params: Dict = field(default_factory=dict)
+
+@dataclass
+class DatasetParams:
+    name: str
+    params: Dict = field(default_factory=dict)
+    transform: Optional[List[AugmentationParams]] = field(default_factory=list)
+    augment: Optional[List[AugmentationParams]] = field(default_factory=list)
+
+@dataclass
+class DataParams:
+    dataset: DatasetParams
+    dataloader: Dict = field(default_factory=dict)
+
+@dataclass
+class DataloaderParams:
+    # I think it must be list, with Enum Phase inside DataParams
+    train: Optional[List[DataParams]]
+    valid: Optional[List[DataParams]]
+    test: Optional[List[DataParams]]
+    predict: Optional[List[DataParams]]
+
+
+# Losses parameters
+@dataclass
+class LossParams:
+    name: str
+    tag: str
+    mapping: Dict[str, str]
+    params: Dict = field(default_factory=dict)
+    weight: Optional[float] = None
+
+@dataclass
+class JointLossParams:
+    loss_params: List[LossParams]
```

Comment:
Let's rename it to `losses` and the outer name in config to `joint_loss`. It might be easier for a user to check if he needs to understand behaviour of the joint loss thing